rapids-cmake/cuda/init_architectures.cmake

```diff
@@ -69,25 +69,14 @@ Example on how to properly use :cmake:command:`rapids_cuda_init_architectures`:
 function(rapids_cuda_init_architectures project_name)
   list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cuda.init_architectures")
 
-  include(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/detail/architectures_policy.cmake)
   # If `CMAKE_CUDA_ARCHITECTURES` is not defined, build for all supported architectures. If
   # `CMAKE_CUDA_ARCHITECTURES` is set to an empty string (""), build for only the current
   # architecture. If `CMAKE_CUDA_ARCHITECTURES` is specified by the user, use user setting.
-  if(DEFINED ENV{CUDAARCHS} AND ("$ENV{CUDAARCHS}" STREQUAL "RAPIDS" OR "$ENV{CUDAARCHS}" STREQUAL
-                                                                        "ALL"))
-    set(cuda_arch_mode "$ENV{CUDAARCHS}")
-    rapids_cuda_architectures_policy(FROM_INIT cuda_arch_mode)
-  elseif(DEFINED ENV{CUDAARCHS} AND "$ENV{CUDAARCHS}" STREQUAL "NATIVE")
-    set(cuda_arch_mode "NATIVE")
-  elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "RAPIDS" OR CMAKE_CUDA_ARCHITECTURES STREQUAL "ALL")
+  if(CMAKE_CUDA_ARCHITECTURES STREQUAL "RAPIDS" OR CMAKE_CUDA_ARCHITECTURES STREQUAL "NATIVE")
     set(cuda_arch_mode "${CMAKE_CUDA_ARCHITECTURES}")
-    rapids_cuda_architectures_policy(FROM_INIT cuda_arch_mode)
-  elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
-    set(cuda_arch_mode "NATIVE")
-    set(deprecated_cuda_arch_mode "EMPTY_STR")
-    rapids_cuda_architectures_policy(FROM_INIT deprecated_cuda_arch_mode)
-  elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "NATIVE")
-    set(cuda_arch_mode "NATIVE")
+  elseif(DEFINED ENV{CUDAARCHS} AND ("$ENV{CUDAARCHS}" STREQUAL "RAPIDS" OR "$ENV{CUDAARCHS}"
```

Comment:
Have an outstanding issue where we need to only evaluate this env variable on the first configuration of a project.